.github/workflows/ci.yml

```diff
@@ -11,24 +10,105 @@ on:
-        run: |
-          mvn -U -B install -DskipTests -Dmaven.javadoc.skip -Dmaven.site.skip -pl jsonschema2pojo-gradle-plugin -am
-          mvn -U -B install -pl jsonschema2pojo-gradle-plugin
+        run: ./mvnw verify -B
+
+      - name: Prepare for next jobs - Install current SNAPSHOT
+        run: ./mvnw -U -B install -DskipTests -Dmaven.javadoc.skip -Dmaven.site.skip
+
+      - name: Prepare for next jobs - Upload org.jsonschema2pojo artifacts (per JDK)
+        uses: actions/upload-artifact@v4
+        with:
+          name: m2-jsonschema2pojo-jdk${{ matrix.java }}
+          path: ~/.m2/repository/org/jsonschema2pojo
+          if-no-files-found: error
+
+  sample-android:
+    needs: project-test
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      fail-fast: false
+      matrix:
+        build-java: [8, 11, 17, 21]
+        run:
+          - { java: 8,  gradle: '7.6.4' }  # AGP Java 8 => Gradle 7.6.4 (MavenPlugin legacy)
+          - { java: 11, gradle: '7.6.4' }
+          - { java: 17, gradle: '8.9'  }
+          - { java: 21, gradle: '8.9'  }
+    name: JDK ${{ matrix.build-java }} | Sample Android | run on Java ${{ matrix.run.java }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Android SDK
+        uses: malinskiy/action-android/install-sdk@release/0.1.1
+
+      - name: Download m2 org.jsonschema2pojo (matching build-java)
+        uses: actions/download-artifact@v4
+        with:
+          name: m2-jsonschema2pojo-jdk${{ matrix.build-java }}
+          path: ~/.m2/repository/org/jsonschema2pojo
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.run.java }}
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+        with:
+          gradle-version: ${{ matrix.run.gradle }}
+
+      - name: Build Android sample
+        working-directory: jsonschema2pojo-gradle-plugin/example/android
+        run: gradle --warning-mode all --info assembleDebug
+
+  sample-java:
+    needs: project-test
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      fail-fast: false
+      matrix:
+        build-java: [8, 11, 17, 21]
+        run-java: [8, 11, 17, 21]
+    name: JDK ${{ matrix.build-java }} | Sample Java | run on Java ${{ matrix.run-java }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Download m2 org.jsonschema2pojo (matching build-java)
+        uses: actions/download-artifact@v4
+        with:
+          name: m2-jsonschema2pojo-jdk${{ matrix.build-java }}
+          path: ~/.m2/repository/org/jsonschema2pojo
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.run-java }}
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+        with:
+          gradle-version: 8.9
+
+      - name: Build Java sample
```

Comment:
Use `Gradle` instead of `Java` here as well.

Reply:
I see this was updated.  Thanks!

Reply:
I've inline with the folder name: `sample` -> `example`